crates/burn-tensor/src/tensor/api/float.rs

```diff
@@ -261,4 +264,33 @@ where
             .matmul(centered)
             .div_scalar(n as f32 - correction_factor as f32)
     }
+
+    /// Sort the elements by value in ascending order along a given dimension.
+    ///
+    /// This sort is unstable (i.e., may reorder equal elements).
+    #[cfg(all(not(feature = "wasm-sync"), target_family = "wasm"))]
+    pub async fn sort(self, dim: usize) -> Tensor<B, D> {
+        Tensor::new(sort::<B, D, Float>(self.primitive, dim, /*descending*/ false).await)
+    }
+
+    /// Sort the elements by value in ascending order along a given dimension.
+    /// Also returns the indices.
+    ///
+    /// This sort is unstable (i.e., may reorder equal elements).
+    #[cfg(all(not(feature = "wasm-sync"), target_family = "wasm"))]
+    pub async fn sort_with_indices(self, dim: usize) -> (Tensor<B, D>, Tensor<B, D, Int>) {
+        check!(TensorCheck::sort_dim::<D>("Sort_with_indices", dim));
+        let (values, indices) =
+            sort_with_indices::<B, D, Float>(self.primitive, dim, /*descending*/ false).await;
+        (Tensor::new(values), Tensor::new(indices))
+    }
+
+    /// Returns the indices that sort the elements by value in ascending order along a given dimension.
+    ///
+    /// This sort is unstable (i.e., may reorder equal elements).
+    #[cfg(all(not(feature = "wasm-sync"), target_family = "wasm"))]
+    pub async fn argsort(self, dim: usize) -> Tensor<B, D, Int> {
```

Comment:
Could we put those into `numeric` to avoid duplication? With newer version of Rust, we can add `async` in traits.

Reply:
Also, what is the reason we need an async API here?

Reply:
Oh I guess I should update my rust version, it wouldn't let me do it initially.

Reply:
Nevermind it was a warning not an error 🤦 

```
use of `async fn` in public traits is discouraged as auto trait bounds cannot be specified
```

Reply:
Will leave it like this for now to avoid the warnings. This will be temporary anyways with #1490.